src/main/java/games/strategy/engine/config/PropertyFileReader.java

```diff
@@ -27,14 +27,22 @@
 
 
   String readProperty(final String propertyKey) {
+    return readProperty(propertyKey, true);
+  }
+
+  String readProperty(final String propertyKey, boolean throwException) {
```

Comment:
I believe this method was added so you could handle the case of the `experimental_ui` property not being present in `game_engine.properties`.  That actually seems like it could be a common use case.  I'm not sure why `PropertyFileReader` only handles this case by throwing an exception.  Returning `Optional<String>` is another possibility.  Or if the semantics of `readProperty()` are such that we need to keep throwing the exception, then maybe introduce a different method that doesn't throw (e.g. `tryReadProperty()`).  However, this is probably a deeper issue and @DanVanAtta should comment on how he sees `PropertyFileReader` evolving in the future.

Another possibility is to change the overload to take a default instead of the `throwException` parameter.  Then you can simply return the default value instead of `null`.

If you choose to keep things as-is, please at least document that if the property is not found, and if `throwException` is `false`, then the method will return `null`.